### PR TITLE
nc_news_6

### PR DESCRIPTION
**Pull request by theRohirrim**
completes happy and sad path tests and functionality for GET /api/articles/:article_id/comments. Endpoints.json have been updated

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -140,4 +141,61 @@ describe("GET /api", () => {
...
+            expect(Array.isArray(res.body.comments)).toBe(true);
+        });
+    })
+
+    test("200: Comments should have correct properties", () => {
+        return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then((res) => {
+            res.body.comments.forEach((comment) => {
```
Comment (itsaliabdallah) [inline __tests__/app.test.js:160]: If you are forEaching through an array you need to do a length check, if an empty array comes back this test will pass.

File: models/topics.model.js [hunk 2]
```diff
@@ -23,4 +23,18 @@ exports.selectEndpoints = () => {
...
+exports.selectCommentsByArticleId = (article_id) => {
+    return db.query(`SELECT comment_id, comments.votes, comments.created_at, comments.author, comments.body, article_id
+                     FROM comments 
+                     JOIN articles USING (article_id)
+                     WHERE article_id = $1
+                     ORDER BY comments.created_at DESC;`, [article_id])
+    .then(({rows}) => {
+        if (rows.length === 0) {
+            return Promise.reject({status: 404, msg: "there are no comments for this article"})
+        }
```
Comment (itsaliabdallah) [inline models/topics.model.js:37]: If there are no comments for the article it is not an error and we would want to send the empty array back to the client.

**Review by itsaliabdallah — CHANGES_REQUESTED** [review]
Please could you add a test for what if your array comes empty we should get a 200 as there is no comments but it's not an error.

Look at the errors PSQL give back if the article ID is not a number or doesn't exist.

Please can you include this test and refactor and submit again.
Thanks

**Review by itsaliabdallah — CHANGES_REQUESTED** [review]
Please could you add a test for what if your array comes empty we should get a 200 as there is no comments but it's not an error.

Look at the errors PSQL give back if the article ID is not a number or doesn't exist.

Please can you include this test and refactor and submit again.
Thanks

**Review by itsaliabdallah — APPROVED** [review]
Awesome job implementing the feedback.
Merge when ready.
